src/formatDate.js

```diff
@@ -50,7 +50,63 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  // verifying icoming formats
+  if (fromFormat.length !== 4) {
+    throw new Error('Incorrect date format (fromFormat).');
+  }
+
+  if (toFormat.length !== 4) {
+    throw new Error('Incorrect date format (fromFormat).');
+  }
```

Comment:
Nice safety feature, but I'm not sure if it's necessary because each array in the tests has a length of 4